ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7934,6 +7958,45 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
     return output;
   }
 
+  private Path getCTASDestinationTableLocation(CreateTableDesc tblDesc, boolean enableSuffixing) throws SemanticException {
```

Comment:
this part looks similar to what is done in create table. Did we set CTAS path here before or we went through the create table flow?

Reply:
Create table flow happens at the DDLTask which is after the query is analyzed, compiled & data is written in CTAS AFAIK. Hence this path is set before the create table flow. This path is used by the FileSinkOperators to write the data.